SConstruct

```diff
@@ -1,49 +1,59 @@
-import errno, os, shutil
-
-user_libs = ['libboost_date_time.so.1.54.0',
-             'libboost_filesystem.so.1.54.0',
-             'libboost_iostreams.so.1.54.0',
-             'libboost_regex.so.1.54.0',
-             'libboost_system.so.1.54.0',
-             'libxerces-c-3.1.so']
+import os
+import shutil
+import subprocess
+import sys
+
+linux_boost_version = '1.54.0'
+
+libs = ['libboost_date_time',
+             'libboost_filesystem',
+             'libboost_iostreams',
+             'libboost_regex',
+             'libboost_system']
+
+linkflags = []
+if sys.platform == 'linux2':
+    libs += ['libxerces-c-3.1']
+    linkflags = ['-Wl,--gc-sections', '-Wl,-rpath', '.']
+if sys.platform == 'darwin':
```

Comment:
Should this be elif and also a catch-all else at the end to report an unsupported platform?

Reply:
I'll add elif and else raise but I don't think we gain that much - the likelihood of compiling anywhere else is very slim and the previous script didn't raise if Linux wasn't the platform being targetted.